account/identity.go

```diff
@@ -166,13 +166,16 @@ func (m *GormIdentityRepository) Create(ctx context.Context, model *Identity) er
 
 // Lookup looks for an existing identity with the given `profileURL` or creates a new one
 func (m *GormIdentityRepository) Lookup(ctx context.Context, username, profileURL, providerType string) (*Identity, error) {
+	if username == "" || profileURL == "" {
```

Comment:
You still think it doesn't make sense to check if providerType is "" when you are later creating the identity?